src/app/plugins/pubsub.plugin.ts

```diff
@@ -1,23 +1,142 @@
+import Constants from '../../common/constants';
 import { Plugin } from '../../common/plugin';
-import { IContainer, IMessage, ChannelType } from '../../common/types';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+import { MessageEmbed } from 'discord.js';
 
 export class PubSubPlugin extends Plugin {
   public name: string = 'Pub Sub Plugin';
-  public description: string = 'Are publix chicken tender subs on sale?';
-  public usage: string = 'pubsub';
-  public pluginAlias = [];
-  public permission: ChannelType = ChannelType.Bot;
+  public description: string = 'Get prices and steamy pictures of the subs you need in your life.';
+  public usage: string = 'pubSub <subName (optional)> | <"list" | "types" (optional)>';
+  public pluginAlias = ['sub', 'subs', 'sandwich', 'samwich', 'sarnie'];
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Food;
+
+  private _API_URL: string = 'https://api.pubsub-api.dev';
+  private _VALID_KEYS: string[] = ['random', 'list', 'types', ''];
+  private _DEFAULT_INPUT: string = 'chicken-tenders';
+  private _SUBS: string[] = [];
+  private _EMBED_LIST = new MessageEmbed();
+
+  private _SUB_UPD_THRESH: number = 1000 * 60 * 60 * 24; // in ms, one day.
```

Comment:
maybe we start using?

https://nodes.duniter.io/typescript/duniter/typedoc/interfaces/moment.duration.html